### PR TITLE
build: bump angular to 18

### PR DESCRIPTION
**Pull request by madhavilosetty-intel**
…eslint/schematics to 18.0.0

BREAKING CHANGE: bump angular to 18

## PR Checklist

- [ ] Unit Tests have been added for new changes
- [ ] API tests have been updated if applicable
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

### CODE REVIEW

File: sol/src/sol.module.ts [hunk 1]
```diff
@@ -14,15 +14,13 @@ import { TerminalComponent } from './terminal/terminal.component'
...
-  imports: [
-    HttpClientModule,
-    CommonModule,
-    BrowserModule
-  ],
   exports: [
     SolComponent
   ],
-  schemas: [CUSTOM_ELEMENTS_SCHEMA]
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
```
Comment (rsdmike) [inline sol/src/sol.module.ts:20]: Do we still need CUSTOMS_ELEMENTS_SCHEMA?

File: tsconfig.json [hunk 2]
```diff
@@ -5,8 +5,8 @@
     "baseUrl": "./",
     "outDir": "./dist/out-tsc",
     "sourceMap": true,
+    "esModuleInterop": true,
```
Comment (rsdmike) [inline tsconfig.json:8]: Why do we need this?

**Review by rsdmike — CHANGES_REQUESTED** [review]
Can we move these to standalone modules as part of the migration to Angular 18, would co-incide well with a breaking change. 

Maybe rework commit message a bit as it is getting cut off in the title

File: sol/src/sol.module.ts (hunk 1 above)
Comment (madhavilosetty-intel) [reply sol/src/sol.module.ts]: done

**Review by rsdmike — APPROVED** [review]

**Pull request merged**